hex/lib/dependabot/hex/file_fetcher.rb

```diff
@@ -37,36 +39,41 @@ def fetch_files
 
       private
 
+      sig { returns(T.nilable(DependencyFile)) }
       def mixfile
-        @mixfile ||= fetch_file_from_host("mix.exs")
+        @mixfile ||= T.let(fetch_file_from_host("mix.exs"), T.nilable(Dependabot::DependencyFile))
+        fetch_file_from_host("mix.exs")
```

Comment:
🔎 I assume memoization is being used here because `fetch_file_from_host` is expensive. Line 44 does this memoization, but the same method is called again on line  45 (negating memoization)